integration/simulation/simulation_tester.go

```diff
@@ -6,23 +6,20 @@ import (
 	"time"
 
 	"github.com/google/uuid"
+	"github.com/obscuronet/obscuro-playground/go/obscuronode/host"
+	"github.com/obscuronet/obscuro-playground/integration/ethereummock"
 )
 
-// This test creates a network of in memory L1 and L2 nodes, then injects transactions, and finally checks the resulting output blockchain.
-// Running it long enough with various parameters will test many corner cases without having to explicitly write individual tests for them.
-// The unit of time is the "avgBlockDurationUSecs" - which is the average time between L1 blocks, which are the carriers of rollups.
-// Everything else is reported to this value. This number has to be adjusted in conjunction with the number of nodes. If it's too low,
-// the CPU usage will be very high during the simulation which might result in inconclusive results.
-func TestInMemoryMonteCarloSimulation(t *testing.T) {
+type createNetworkFunc = func(nrNodes int, avgGossipPeriod uint64, avgBlockDurationUSecs uint64, avgLatency uint64, stats *Stats) ([]*ethereummock.Node, []*host.Node)
+
+// testSimulation encapsulates the shared logic for simulating and testing various types of nodes.
+func testSimulation(t *testing.T, avgBlockDurationUSecs uint64, createNetwork createNetworkFunc, efficiencies EfficiencyThresholds) {
 	// define core test parameters
 	numberOfNodes := 10
```

Comment:
I'd say to move all these core parameters out of there as well. Maybe create a structure for them

Reply:
which would make the "createNetworkFunc" cleaner as well:

```type createNetworkFunc = func(testParams *TestParams, stats *Stats) ...```